coursier.bzl

```diff
@@ -265,16 +277,43 @@ def _get_jq_http_files():
         ])
     return lines
 
+def _add_outdated_files(repository_ctx, artifacts, repositories):
+    repository_ctx.file(
+        "outdated.artifacts",
+        "\n".join(["{}:{}:{}".format(artifact["group"], artifact["artifact"], artifact["version"]) for artifact in artifacts]) + "\n",
+        executable = False,
+    )
+
+    repository_ctx.file(
+        "outdated.repositories",
+        "\n".join([repo["repo_url"] for repo in repositories]) + "\n",
+        executable = False,
+    )
+
+    repository_ctx.template(
+        "outdated.sh",
+        repository_ctx.attr._outdated,
+        {
+            "{repository_name}": repository_ctx.name,
+            "{proxy_opts}": " ".join(_get_java_proxy_args(repository_ctx)),
+        },
+        executable = True,
+    )
+
 def _pinned_coursier_fetch_impl(repository_ctx):
     if not repository_ctx.attr.maven_install_json:
         fail("Please specify the file label to maven_install.json (e.g." +
              "//:maven_install.json).")
 
     _windows_check(repository_ctx)
 
+    repositories = []
+    for repository in repository_ctx.attr.repositories:
+        repositories.append(json_parse(repository))
+
     artifacts = []
-    for a in repository_ctx.attr.artifacts:
-        artifacts.append(json_parse(a))
+    for artifact in repository_ctx.attr.artifacts:
```

Comment:
In the future, please separate refactoring changes from feature changes for a smaller diff :)